localstack-core/localstack/utils/analytics/usage.py

```diff
@@ -49,6 +49,49 @@ def aggregate(self) -> dict:
         return self.state
 
 
+class UsageMultiSetCounter:
+    """
+    Use this counter to count occurrences of unique values for multiple dimensions.
+    This dynamically creates UsageSetCounters and should be used with care (i.e., with limited keys).
+
+    Example:
+
+    my_feature_counter = UsageMultiSetCounter("pipes:invocation")
+    my_feature_counter.record("aws:sqs", "aws:lambda")
+    my_feature_counter.record("aws:sqs", "aws:lambda")
+    my_feature_counter.record("aws:sqs", "aws:stepfunctions")
+    my_feature_counter.record("aws:kinesis", "aws:lambda")
+    my_feature_counter.aggregate() returns:
+    {
+       "aws:sqs": {
+         "aws:lambda": 2,
+         "aws:stepfunctions": 1,
+       },
+       "aws:kinesis": {
+         "aws:lambda": 1
+       }
+    }
+    """
+
+    namespace: str
+    _counters: dict[str, UsageSetCounter]
+
+    def __init__(self, namespace: str):
+        self._counters = {}
+        self.namespace = namespace
+
+    def record(self, key: str, value: str):
+        namespace = f"{self.namespace}:{key}"
+        set_counter = self._counters.setdefault(key, UsageSetCounter(namespace))
```

Comment:
I think using `setdefault` here might lead to an issue: when initializing a `UsageSetCounter`, it will register itself in the register with `collector_registry[namespace] = self`, overriding the namespace. 

I think we should probably not create the entry if `config.DISABLE_EVENTS` is enabled.